lib/utils.pm

```diff
@@ -1022,4 +1023,27 @@ sub svirt_host_basedir {
     return get_var('VIRSH_OPENQA_BASEDIR', '/var/lib');
 }
 
+sub prepare_ssh_localhost_key_login {
+    my ($source_user) = @_;
+    # in case localhost is already inside known_hosts
+    if (script_run('test -e ~/.ssh/known_hosts') == 0) {
+        assert_script_run('ssh-keygen -R localhost');
+    }
+
+    # generate ssh key
+    if (script_run('! test -e ~/.ssh/id_rsa') == 0) {
+        assert_script_run('ssh-keygen -t rsa -N "" -f ~/.ssh/id_rsa');
+    }
+
+    # add key to authorized_keys of root
+    if ($source_user eq 'root') {
+        assert_script_run('mkdir -p ~/.ssh');
+        assert_script_run('cat ~/.ssh/id_rsa.pub | tee -a ~/.ssh/authorized_keys');
```

Comment:
I think this is duplicating code. Why not using always the sudo variant which will work in all cases?

Reply:
Because it doesn't work if I am already root.
Also as openqa actually uses `su -c` insted of `sudo` I wouldn't be able to find the source home as `~` is expanded after changing the user.

Reply:
The expanding of ~ should not be any problems, since the sudo variant is using absolute paths.
If assert_script_sudo is not working when you are already root, then this method should be adapted instead of duplicating code.